chore: lint Dockerfile and add documentation/examples

azgaviperr opened the pull request:
- Applied linting to the Dockerfile for better readability and consistency.
- Updated README.md with additional details. ( lot of it )
- Added multiple docker-compose example files:
    - docker-compose-config-variable-only.yml
    - docker-compose-multiple-servers-env-files.yml
    - docker-compose-multiple-servers-env-variables.yml
    - docker-compose-single-server.yml
- Included an example environment file (env_example) for reference.

azgaviperr committed:
chore: lint Dockerfile and add documentation/examples

- Applied linting to the Dockerfile for better readability and consistency.
- Updated README.md with additional details. ( lot of it )
- Added multiple docker-compose example files:
    - docker-compose-config-variable-only.yml
    - docker-compose-multiple-servers-env-files.yml
    - docker-compose-multiple-servers-env-variables.yml
    - docker-compose-single-server.yml
- Included an example environment file (env_example) for reference.

azgaviperr committed:
Update README.md to improve documentation clarity

azgaviperr commented:
I made this documentation to help few persons. It's made on my notes. If needed I can split it in a docs/ folder to make it more digest.

azgaviperr committed:
Split documentation for easier use and maintainability

- Restructured documentation into separate sections for better organization:
    - Added `docs/advanced-usage/index.md`
    - Added `docs/configuration-guide/index.md`
    - Added `docs/getting-started/index.md`
    - Added `docs/index.md`
    - Added `docs/troubleshooting/index.md`
- Renamed example files for clarity and consistency:
    - Moved and renamed `docker-compose-examples/docker-compose-multiple-servers-env-files.yml` to `examples/docker-compose-multiple-servers-env-files.yml`
    - Moved and renamed `docker-compose-examples/docker-compose-multiple-servers-env-variables.yml` to `examples/docker-compose-multiple-servers-env-variables.yml`
    - Moved and renamed `docker-compose-examples/docker-compose-single-server.yml` to `examples/docker-compose-single-server.yml`
    - Moved and renamed `docker-compose-examples/docker-compose-config-variable-only.yml` to `examples/docker-compose-static-config.yml`
    - Moved `env_example` to `examples/env_example`
- Updated `README.md` and `docker-compose.yml` to reflect these changes.

azgaviperr commented:
Documentation should be much nicer to review and use now. Let me know.

hafometh88 commented on docs/advanced-usage/index.md:
```diff
@@ -0,0 +1,308 @@
+- Use strong passwords for admin and RCON access.
+- Regularly back up your configuration and profile data.
+
+> **Warning**: Be cautious about who you grant access to your Docker host and server ports. Ensuring a minimum level of security awareness is essential to protect your environment from unauthorized access or malicious activity.
```
This warning seems really out of place to me, but I guess why not

azgaviperr replied:
Yeah, just a bad memory experience and people to easily letting people to connect to their hosted server for letting people help and get burned :p

hafometh88 replied:
Yes

azgaviperr replied:
Removed the Warning.

hafometh88 commented on docs/advanced-usage/index.md:
````diff
@@ -0,0 +1,308 @@
+1. Start the server once to generate the default `docker_generated.json` file:
+
+    ```bash
+    docker-compose up
````
```suggestion
    docker compose up
```

You are using a deprecated docker compose v1 syntax.

hafometh88 commented on docs/advanced-usage/index.md:
````diff
@@ -0,0 +1,308 @@
+5. Restart the server to apply the changes:
+
+    ```bash
+    docker-compose down && docker-compose up
````
```suggestion
    docker compose down && docker compose up
```

You are using a deprecated docker compose v1 syntax.

hafometh88 commented on docs/advanced-usage/index.md:
````diff
@@ -0,0 +1,308 @@
+### Example: Docker Compose for Two Servers
+
+```yaml
+version: "3.8"
````
```suggestion
---
```

The version property is deprecated.

azgaviperr replied:
Thanks @hafometh88 missed this one

hafometh88 commented on docs/advanced-usage/index.md:
````diff
@@ -0,0 +1,308 @@
+- **Separate Compose Files**: Alternatively, you can create multiple `docker-compose.yml` files for each server instance. This allows you to run and manage each instance independently. For example:
+
+```bash
+docker-compose -f docker-compose-server1.yml up -d
````
```suggestion
docker compose -f docker-compose-server1.yml up -d
```

hafometh88 commented on docs/advanced-usage/index.md:
````diff
@@ -0,0 +1,308 @@
+
+```bash
+docker-compose -f docker-compose-server1.yml up -d
+docker-compose -f docker-compose-server2.yml up -d
````
```suggestion
docker compose -f docker-compose-server2.yml up -d
```

hafometh88 commented on docs/advanced-usage/index.md:
```diff
@@ -0,0 +1,308 @@
+
+## Additional Server Parameters
+
+You can pass additional parameters to the server binary using the `ARMA_PARAMS` environment variable. This is useful for parameters that do not have specific environment variables assigned. For example:
```
I would provide a link to BI docs so the user can easily see the avaliable startup parameters.

azgaviperr replied:
Good idea !

azgaviperr replied:
Changed the way to handle it

hafometh88 commented on docs/advanced-usage/index.md:
```diff
@@ -0,0 +1,308 @@
+
+## Workshop and Install Folder Sharing
+
+To save disk space and reduce download times, you can share the same workshop folder between multiple server instances by mounting the same host directory to `/reforger/workshop` for each service. Note that some mods may not support concurrent access.
```
I REALLY would NOT do this! The mods can get locked when redownloading and this can lead to unpredictable behaviour.

azgaviperr replied:
That why I add the Note that some mods may not support concurrent access.
But you may be right, maybe a better idea to don't even mention it and let people experiment on their own. Would love other feedbacks

hafometh88 replied:
Advanced users can figure this out on their own and recommending is would just cause potential issues.

azgaviperr replied:
Will remove.

azgaviperr replied:
Removed

hafometh88 commented on docs/getting-started/index.md:
````diff
@@ -0,0 +1,52 @@
+1. Create a `docker-compose.yml` file:
+
+    ```yaml
+    version: "3.9"
````
```suggestion
    ---
```

hafometh88 commented on docs/getting-started/index.md:
````diff
@@ -0,0 +1,52 @@
+2. Start the server:
+
+    ```bash
+    docker-compose up -d
````
```suggestion
    docker compose up -d
```

hafometh88 commented on README.md:
````diff
@@ -1,83 +1,57 @@
 
-A comma separated list of IDs, with an optional version.
+   ```bash
+   docker-compose down
````
```suggestion
   docker compose down
```

hafometh88 commented on README.md:
````diff
@@ -1,83 +1,57 @@
 
-Workshop mods can be defined in two ways. You can use both or either of those.
+   ```bash
+   docker-compose up -d
````
```suggestion
   docker compose up -d
```

hafometh88 reviewed (commented):
Expanding the docs is a good idea, but this has a lot of unnecessary filler in my opinion

azgaviperr committed:
Update docs/advanced-usage/index.md

Co-authored-by: hafometh88 <76481827+hafometh88@users.noreply.github.com>

azgaviperr committed:
Update docs/advanced-usage/index.md

Co-authored-by: hafometh88 <76481827+hafometh88@users.noreply.github.com>

azgaviperr committed:
Update docs/advanced-usage/index.md

Co-authored-by: hafometh88 <76481827+hafometh88@users.noreply.github.com>

azgaviperr committed:
Update docs/advanced-usage/index.md

Co-authored-by: hafometh88 <76481827+hafometh88@users.noreply.github.com>

azgaviperr committed:
Update docs/getting-started/index.md

Co-authored-by: hafometh88 <76481827+hafometh88@users.noreply.github.com>

azgaviperr committed:
Update docs/getting-started/index.md

Co-authored-by: hafometh88 <76481827+hafometh88@users.noreply.github.com>

azgaviperr committed:
Update README.md

Co-authored-by: hafometh88 <76481827+hafometh88@users.noreply.github.com>

azgaviperr committed:
Update README.md

Co-authored-by: hafometh88 <76481827+hafometh88@users.noreply.github.com>

azgaviperr committed:
Update docs/advanced-usage/index.md

Co-authored-by: hafometh88 <76481827+hafometh88@users.noreply.github.com>

azgaviperr committed:
docs: update advanced usage guide with @hafometh88's change proposal

Incorporated suggestions from @hafometh88 to improve clarity and detail in the advanced usage documentation.

azgaviperr committed:
fix: remove unnecessary quotes in port mappings in example compose files

Removed redundant double quotes from port mapping values in the example Docker Compose files for improved clarity and consistency.

azgaviperr commented:
> Expanding the docs is a good idea, but this has a lot of unnecessary filler in my opinion

What do you think about it now ? Any other unnecessary filler ?
